sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,13 +349,28 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import sin, sinh
+        arg = self.args[0]
+
+        # `cosh(x)` is real for real and purely imaginary `x`
+        if arg.is_real or arg.is_imaginary:
+            return True
+
+        # cosh(a+ib) = cos(b)*cosh(a) + i*sin(b)*sinh(a)
+        if arg.is_number:
+            re, im = arg.as_real_imag()
+            if (sin(im)*sinh(re)).equals(0):
```

Comment:
sinh(re) is zero if and only if re is zero, or arg is imaginary. But that was tested already.
sin(im) is zero if and only if im is a multiple of pi. That can be tested by something like  im % pi == 0 when im is a number without having to create  sin(im).

It arg is not a number, then some symbolic form of these tests could be considered.